### PR TITLE
Add obsp and varp

### LINKED ISSUE #136: Backwards breaking change: Sparse observation and variables annotation
We had this discussion several times already, last time here: https://github.com/theislab/anndata/pull/133

It would be great to generate a new attribute for sparse annotations of observations and variables. Presumably `.obss` and `.vars`, where `s` stands for sparse.

I'd assume that @Koncopd could be the one to implement it, let's talk about it soon! If anyone else would feel there is bandwidth, please let me know.

### PR DESCRIPTION
**Pull request by ivirshup**
Fixes #136

This introduces attributes for pairwise measurements on the observations and variables. Currently mostly for cell-cell graphs, but could also be used for gene correlation graphs.

Thoughts @falexwolf @flying-sheep @Koncopd?

### CODE REVIEW

**Comment by flying-sheep** [comment]
Great! We will need code in scanpy that automatically checks for square sparse matrices in `.uns` which will be superfluous after this.

File: anndata/core/anndata.py [hunk 1]
```diff
@@ -655,12 +655,12 @@ def _init_as_view(self, adata_ref: 'AnnData', oidx: Index, vidx: Index):
...
-        # varp=None, obsp=None,
-        raw=None, layers=None,
-        dtype='float32', shape=None,
-        filename=None, filemode=None,
+            self, X=None, obs=None, var=None, uns=None,
+            obsm=None, varm=None,
+            varp=None, obsp=None,
+            raw=None, layers=None,
+            dtype='float32', shape=None,
+            filename=None, filemode=None
```
Comment (flying-sheep) [inline anndata/core/anndata.py]: Please rebase on master so my style edits don’t get reverted

File: anndata/core/alignedmapping.py [hunk 2]
```diff
@@ -292,30 +292,29 @@ def dim(self):
         return self._dimnames[self._axis]
 
 
-
 # TODO: Allow these to be written, and add them back to anndata
```
Comment (flying-sheep) [inline anndata/core/alignedmapping.py]: The TODO won’t be necessary right?

**Comment by ivirshup** [comment]
Which renaming?

**Comment by flying-sheep** [comment]
How the abstract classes and the concrete/view classes are named, as I said in the other PR. This PR introduces more of those which is why I mentioned it.

**Comment by ivirshup** [comment]
Gotcha. I agree some renaming and decisions on exports need to be made. I'm still considering my thoughts on this, but I'll add those to the renaming issue soon.

**Pull request merged**